hls4ml/model/types.py

```diff
@@ -311,16 +314,20 @@ def __eq__(self, other):
         return eq
 
 
-class XnorPrecisionType(IntegerPrecisionType):
+class XnorPrecisionType(PrecisionType):
     """
     Convenience class to differentiate 'regular' integers from BNN Xnor ones
     """
 
     def __init__(self):
         super().__init__(width=1, signed=False)
```

Comment:
Logically xnor is signed, even if we represent it as unit<1>. How it gets implemented should be a backend issue, not a definition issue.

Reply:
I think the XNOR idea was that it is a single bit implementation, the more general definition of BNN is signed (-1 and 1), but that's a different case.